params/config.go

```diff
@@ -476,6 +476,36 @@ var (
 		Consortium:                    nil,
 		ConsortiumV2Contracts:         nil,
 	}
+	NonActivatedConfig = &ChainConfig{
```

Comment:
Moving genesis init to `NewBlockChain` results in more checking when calling it. `TestChainConfig` currently turns on a lot of fork blocks at block 0, which makes some tests to fail. `NonActivatedConfig` is added with all features turned off so that we can adjust manually for each test.

Reply:
@Francesco4203 so as I understood, moving this logic will make some test trigger the logic genesis init step here, then  when using param.TestChainConfig it will be compatiable with current test, then we need to reinit new config for testing right ? Why does we have the the custom failed here comparing to  PR in ethereum, and turning off cnfig will skip some testing checking or not ?

Reply:
@huyngopt1994 
- Yes.

>  Why does we have the the custom failed here comparing to PR in ethereum
- What custom failed are you referring to?
>  and turning off cnfig will skip some testing checking or not ?
- For now, this config is only used in `core/dao_test.go/TestDAOForkRangeExtradata`, in which we turning off all configs and only turn on the DAO fork block. The reason for this is that in that test case, the DAO fork block is set at  32, not 0 as in the `TestChainConfig`, so we have to set it manually. The checking in genesis init will still happen. Not sure if this `NonActivatedConfig` is going to be used for more test in the future.

Reply:
> What custom failed are you referring to?

Ah I means that Why does Ethereum encounter this issue like us. Any reason behind

Reply:
Ok I found some logics changed from the original PR.